main.tf

```diff
@@ -0,0 +1,60 @@
+provider "vault" {} // Ensure VAULT_ADDR and VAULT_TOKEN env vars are set
+
+resource "vault_mount" "database" {
+  path                      = "database"
+  type                      = "database"
+  default_lease_ttl_seconds = "${local.database["lease_ttl"]}"
+  max_lease_ttl_seconds     = "${local.database["lease_ttl"]}"
+}
+
+resource "vault_database_secret_backend_connection" "postgres" {
+  allowed_roles = ["service-write", "dev-read"]
+  backend       = "${vault_mount.database.path}"
+  name          = "postgres-secret-backend"
+
+  // Needed as SSL is disabled on postgres Docker container
+  verify_connection = false
+
+  postgresql {
+    // sslmode=disable needed for the Postgres Go Client
+    connection_url = "postgres://${var.postgres_db_user}:${var.postgres_db_password}@${var.postgres_db_endpoint}/${var.postgres_db}?sslmode=disable"
```

Comment:
Maybe drop the `postgres_` prefix on these variables? Would make this line a lot shorter.